### PR TITLE
SETI-1335: Add possibility to config keepalive for gearman server

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/zuul-opensource-repository-check/20/ : SUCCESS in 14s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/187/ : SUCCESS in 15m 05s

**Comment by michalvanco** [comment]
is there any place with related tests?

**Comment by janpriessnitz** [comment]
Not really. Either I can create new tests for internal gearman server config, or test it manually. What do you think?

File: zuul/cmd/server.py [hunk 1]
```diff
@@ -128,11 +128,16 @@ def start_gear_server(self):
                 host = self.config.get('gearman_server', 'listen_address')
             else:
                 host = None
+            if self.config.has_option('gearman_server', 'keepalive'):
+                keepalive = self.config.get('gearman_server', 'keepalive')
```
Comment (hrubi) [inline zuul/cmd/server.py]: You want `getboolean` instead. In `gear.Server` there's `if keepalive:`, so having some nonempty string value retrieved with `get` would be always true.

File: doc/source/zuul.rst [hunk 2]
```diff
@@ -75,6 +75,10 @@ than connecting to an external one.
   Path to log config file for internal Gearman server.
   ``log_config=/etc/zuul/gearman-logging.yaml``
 
+**keepalive**
+  Whether gearman should send keepalive packets in connections to its workers.
```
Comment (hrubi) [inline doc/source/zuul.rst]: Specify the default value as it is done for the options above.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/zuul-opensource-repository-check/24/ : SUCCESS in 7s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/195/ : SUCCESS in 7m 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/zuul-opensource-repository-check/26/ : SUCCESS in 7s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/197/ : SUCCESS in 7m 00s

**Comment by hrubi** [comment]
ad tests: add a test into tests/test_scheduler.py which will take a config with keepalive=true, run a job for a project and check that it succeeded. basically just copy some simple test and let it use the new config with keepalive.

**Comment by hrubi** [comment]
ad tests: ok, so we decided with @janpriessnitz that it's not worth it. the gear server is started differently in prod and in tests. so we'll rely on integration testing here.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline gate aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/zuul-merge-compliance-check/8/ : SUCCESS in 18s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/204/ : SUCCESS in 9m 51s

**Pull request merged**